plugins/qtosgrave/osglodlabel.cpp

```diff
@@ -0,0 +1,102 @@
+    *           |
+    *           |
+    * [Label Offset Transform (optional)]
+    *           |
+    *           |
+    * [Global LOD (this + controls label visibility)]
+    *           |
+    *           |
+    *     [Label Geode]
+    *           |
+    *           |
+    *     [Label Text]
+    */
+
+    // Create text element
+    osg::ref_ptr<osgText::Text> text = new osgText::Text();
+    std::istringstream fontStream(fontString);
+    text->setFont(osgText::readFontStream(fontStream));
+    // Set up other text element properties
+    text->setText(label);
+    text->setCharacterSize(0.05);
+    text->setAutoRotateToScreen(true);
+    text->setPosition( osg::Vec3( 0.0, 0.0, 0.0 ) );
+    text->setDrawMode( osgText::Text::TEXT );
+    text->setColor(osg::Vec4(0,0,0,1));
+    text->setBackdropColor( osg::Vec4( 1.0, 1.0f, 1.0f, 1.0f ) );
+    text->setBackdropType( osgText::Text::OUTLINE );
+    text->setAlignment( osgText::Text::CENTER_CENTER );
+    text->setAxisAlignment( osgText::Text::SCREEN );
+    text->setCharacterSizeMode( osgText::Text::OBJECT_COORDS_WITH_MAXIMUM_SCREEN_SIZE_CAPPED_BY_FONT_HEIGHT );
+    text->setMaximumHeight(0.5);
+    text->setFontResolution(128,128);
+
+    // Override the cartoon shader with the default shader for text
+    osg::ref_ptr<osg::Program> program = new osg::Program;
+    text->getOrCreateStateSet()->setAttributeAndModes(program.get(), osg::StateAttribute::PROTECTED | osg::StateAttribute::ON);
+
+    // Ensure that the text is drawn over any shape in the scene
+    text->getOrCreateStateSet()->setMode(GL_LIGHTING, osg::StateAttribute::PROTECTED | osg::StateAttribute::OFF );
+    text->getOrCreateStateSet()->setAttributeAndModes( new osg::Depth(osg::Depth::ALWAYS), osg::StateAttribute::PROTECTED | osg::StateAttribute::ON);
+    text->getOrCreateStateSet()->setMode(GL_DEPTH_TEST, osg::StateAttribute::PROTECTED | osg::StateAttribute::OFF );
+
+    // Place text element into geode node
+    osg::ref_ptr<osg::Geode> textGeode = new osg::Geode();
+    textGeode->addDrawable(text);
+
+    // Place geode node into LOD node for distance-based culling
+    this->addChild(textGeode, 0, 20);
+}
+
+// Debug print during label destruction for testing
+OSGLODLabel::~OSGLODLabel() {
+    // RAVELOG_DEBUG("Destroying OSGLODLabel!");
+}
+
+// Overriding of default LOD node traverse
+void OSGLODLabel::traverse(osg::NodeVisitor& nv)
+{
+    // Use global distance instead of local distance from camera if node visitor settings are applicable
+    // Otherwise, use the default LOD node implementation of this function
+    if (nv.getTraversalMode() == osg::NodeVisitor::TRAVERSE_ACTIVE_CHILDREN && _rangeMode==DISTANCE_FROM_EYE_POINT) {
+        float requiredRange = nv.getDistanceToViewPoint(getCenter(),true);
+        // The input node visitor, represented as a CullStack
+        osg::CullStack *cullStack = dynamic_cast<osg::CullStack *>(&nv);
+        if (cullStack != NULL) {
```

Comment:
better to use `nullptr`?

Reply:
yes, actually I also commented about that, not sure where the comment is though.